esmvaltool/recipes/recipe_rainfarm.yml

```diff
@@ -0,0 +1,49 @@
+# recipe_rainfarm.yml
+---
+
+documentation:
+  description: |
+    Recipe for performing stochastic downscaling of precipitation fields
+    calling the RainFARM package by J. von Hardenberg (ISAC-CNR)
+  authors:
+    - arno_en
+    - hard_jo
+
+  references:
+    - donofrio14jh
+    - rebora06jhm
+
+  projects:
+    - c3s-magic
+
+datasets:
+  - {dataset: EC-EARTH,       project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 1997, end_year: 1997}
+
+preprocessors:
+  preproc:
+    extract_levels: false
+
+diagnostics:
+  rainfarm:
+    description: RainFARM stochastic downscaling of precipitation fields
+    variables:
+      pr:
+        preprocessor: preproc
+        reference_dataset: "EC-EARTH"
+        mip: day
+        field: T2Ds
+    scripts:
+      main:
+        script: rainfarm/main.r
+        # Diagnostic arguments: set FALSE to void and allow RainFARM automated setting
+        force_processing: true      # TRUE to rewrite existing output file
+        rlonlatdata: [4,13,44,53]   # Subsetting of region where calculation is to be performed. 
```

Comment:
Note that the esmvaltool preprocessor supports both area selection and regridding. If possible, use that instead of doing it in the diagnostic, that will be much quicker since it avoids copying a large amount of data to the preprocessor directory.

To use this, you would define the preprocessor:
```yaml
preprocessors:
  preproc:
    extract_region:
      start_longitude: 3
      end_longitude: 13
      start_latitude: 44
      end_latitude: 53
    regrid:
      target_grid: 1x1  # in degrees, can also be the name of one of the datasets to use the grid from that dataset
      scheme: linear  # can be linear, nearest, area_weighted, unstructured_nearest
```

Reply:
I used the area selection but then was told by @valeriupredoi not to as it was untested as yet (sorry, I cannot find the issue anymore). Is this ok now? Also, when using it the area selection does not return results within requested boundaries but also outside it - I guess it works on the nearest point? This can be unexpected and can cause troubles. Can you check this?

Sure, will switch regridding to preproc. Lost it when removing the area selection.

Reply:
the `esmvaltool.preprocessor._area_pp._area_slice()` should work OK now, it has been tested by @ledm in his recipes and we didn't get any complaints - I haven't tested it myself but if you say it returns points outside the box I guess it is, as you say, beacuse of next available point optimization

Reply:
looking at `cube.intersection()` it looks like one can specify the min/max ranges to include as keyword arguments: https://scitools.org.uk/iris/docs/v1.9.0/html/iris/iris/cube.html

Reply:
Would it be worth adding a "scheme" variable to the `area_slice` function?

Reply:
I switched area selection to the pre-proc. Indeed @ledm a "scheme" variable for this would be good too. Or @valeriupredoi, any other way to control how the selection is performed.